homeassistant/components/zwave_js/config_flow.py

```diff
@@ -309,6 +329,71 @@ async def async_step_user(
 
         return await self.async_step_manual()
 
+    async def async_step_usb(self, discovery_info: dict[str, str]) -> FlowResult:
+        """Handle USB Discovery."""
+        if not is_hassio(self.hass):
+            return self.async_abort(reason="discovery_requires_supervisor")
+        if self._async_current_entries():
+            return self.async_abort(reason="already_configured")
+        if self._async_in_progress():
+            return self.async_abort(reason="already_in_progress")
+
+        vid = discovery_info["vid"]
+        pid = discovery_info["pid"]
+        serial_number = discovery_info["serial_number"]
+        device = discovery_info["device"]
+        manufacturer = discovery_info["manufacturer"]
+        description = discovery_info["description"]
+        # The Nortek sticks are a special case since they
+        # have a Z-Wave and a Zigbee radio. We need to reject
+        # the Zigbee radio.
+        if (
+            vid == "10C4"
+            and pid == "8A2A"
+            and "Z-Wave" not in discovery_info["description"]
+        ):
+            return self.async_abort(reason="not_zwave_device")
+        # Zooz uses this vid/pid, but so do 2652 sticks
+        if vid == "10C4" and pid == "EA60" and "2652" in discovery_info["description"]:
+            return self.async_abort(reason="not_zwave_device")
+
+        addon_info = await self._async_get_addon_info()
+        if addon_info.state not in (AddonState.NOT_INSTALLED, AddonState.NOT_RUNNING):
+            return self.async_abort(reason="already_configured")
+
+        await self.async_set_unique_id(
+            f"{vid}:{pid}_{serial_number}_{manufacturer}_{description}"
+        )
+        self._abort_if_unique_id_configured()
+        dev_path = await self.hass.async_add_executor_job(get_serial_by_id, device)
+        self.usb_path = dev_path
+        self._title = _format_port_human_readable(
+            dev_path,
+            serial_number,
+            manufacturer,
+            description,
+            vid,
+            pid,
+        )
+        self.context["title_placeholders"] = {CONF_NAME: self._title}
+        return await self.async_step_usb_confirm()
+
+    async def async_step_usb_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle USB Discovery confirmation."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="usb_confirm",
+                description_placeholders={CONF_NAME: self._title},
+                data_schema=vol.Schema({}),
+            )
+
+        addon_info = await self._async_get_addon_info()
```

Comment:
Maybe we can just go to `async_step_on_supervisor` here instead, and passing `CONF_USE_ADDON: True`?

Reply:
I think so. testing it now

Reply:
uninstalled addon, deleted config entry, restarted with the change, discovered, proceeded to set it up and it worked like a charm